### PR TITLE
additional error handling and tests

### PR DESCRIPTION
**Pull request by SKopecz**
Added missing error checks for MPRK22, MPRK43I and MPRK43II.

### CODE REVIEW

File: src/mprk.jl [hunk 1]
```diff
@@ -501,6 +501,10 @@ function alg_cache(alg::MPRK22, u, rate_prototype, ::Type{uEltypeNoUnits},
                    ::Type{uBottomEltypeNoUnits}, ::Type{tTypeNoUnits},
                    uprev, uprev2, f, t, dt, reltol, p, calck,
                    ::Val{true}) where {uEltypeNoUnits, uBottomEltypeNoUnits, tTypeNoUnits}
+    if !(f isa PDSFunction || f isa ConservativePDSFunction)
+        throw(ArgumentError("MPRK22 can only be applied to production-destruction systems"))
+    end
+    @assert alg.alpha≥1 / 2 "MPRK22 requires α ≥ 1/2."
```
Comment (ranocha) [inline src/mprk.jl:507]:
```suggestion
    if !(alg.alpha ≥ 1 / 2)
        throw(ArgumentError("MPRK22 requires α ≥ 1/2."))
    end
```

From the docstring:
```julia
help?> @assert
  @assert cond [text]

  Throw an AssertionError if cond is false. Preferred syntax for writing assertions. Message text
  is optionally displayed upon assertion failure.

  │ Warning
  │
  │  An assert might be disabled at various optimization levels. Assert should therefore
  │  only be used as a debugging tool and not used for authentication verification (e.g.,
  │  verifying passwords), nor should side effects needed for the function to work
  │  correctly be used inside of asserts.

```

Thus, I think it's better to throw a proper error in this case.

File: src/mprk.jl [hunk 2]
```diff
@@ -542,9 +546,10 @@ function alg_cache(alg::MPRK22, u, rate_prototype, ::Type{uEltypeNoUnits},
                    ::Type{uBottomEltypeNoUnits}, ::Type{tTypeNoUnits},
                    uprev, uprev2, f, t, dt, reltol, p, calck,
                    ::Val{false}) where {uEltypeNoUnits, uBottomEltypeNoUnits, tTypeNoUnits}
-
-    #TODO: Should assert alg.alpha >= 0.5
-
+    if !(f isa PDSFunction || f isa ConservativePDSFunction)
+        throw(ArgumentError("MPRK22 can only be applied to production-destruction systems"))
+    end
+    @assert alg.alpha≥1 / 2 "MPRK22 requires α ≥ 1/2."
```
Comment (ranocha) [inline src/mprk.jl:552]:
```suggestion
    if !(alg.alpha ≥ 1 / 2)
        throw(ArgumentError("MPRK22 requires α ≥ 1/2."))
    end
```

File: test/runtests.jl [hunk 3]
```diff
@@ -286,8 +286,22 @@ const prob_pds_linmod_nonconservative_inplace = PDSProblem(linmodP!, linmodD!, [
...
                                                                                            MPE(),
                                                                                            dt = 0.25)
-            #TODO: Test that MPRK22 requires α ≥ 1/2 (not yet implemented)
-            #TODO: Test that MPRK22, MPRK43I, MPRK43II can only be applied to PDS (not yet implemented)
+            @test_throws "MPRK22 can only be applied to production-destruction systems" solve(prob_oop,
+                                                                                              MPRK22(1.0))
+            @test_throws "MPRK22 can only be applied to production-destruction systems" solve(prob_ip,
+                                                                                              MPRK22(1.0))
+            @test_throws "MPRK22 requires α ≥ 1/2." solve(prob_pds_linmod, MPRK22(0.25))
+            @test_throws "MPRK22 requires α ≥ 1/2." solve(prob_pds_linmod, MPRK22(0.25))
```
Comment (ranocha) [inline test/runtests.jl:294]: I guess one of these tests should be for the in-place version and the other one for the out-of-place version?

**Review by ranocha — CHANGES_REQUESTED** [review]
Thanks! This looks already quite good to me. Please find some comments below.

File: test/runtests.jl (hunk 3 above)
Comment (SKopecz) [reply test/runtests.jl:294]: You're right!

Comment (SKopecz) [reply test/runtests.jl:294]: I'll add a function `get_constant_parameters` for `MPRK22`, just like for `MPRK43`. This way the constant parameters are the same for in-place and out-of-place implementations and it is sufficient to test the out-of-place version.

**Comment by SKopecz** [comment]
Now all `@assert`s writing a text should have been replaced by `throw(ArgumentError())`.

**Review by ranocha — APPROVED** [review]
Thanks!

**Pull request merged**